forward_engineering/services/statements/insertStatements.js

```diff
@@ -0,0 +1,58 @@
+const _ = require('lodash');
+const uuid = require('uuid');
+const { getKeySpaceReference } = require('./commonStatements');
+
+/**
+ *
+ * @param {jsonData: object, collections: object[]} param
+ * @returns {string[]}
+ */
+const getInsertScripts = ({ jsonData, collections = [] }) => {
+	const collectionsScripts = collections
+		.map(collection => {
+			const collectionJsonData = jsonData[collection.GUID];
+			return getInsertScriptForCollection({
+				jsonData: collectionJsonData,
+				collection,
+			});
+		})
+		.join('\n\n');
+
+	return collectionsScripts;
+};
+
+/**
+ *
+ * @param {jsonData: object, collection: object} param0
+ * @returns {string}
+ */
+const getInsertScriptForCollection = ({ jsonData, collection }) => {
+	if (collection.isActivated === false) {
+		return '';
+	}
+
+	const keyPropertyName = Object.keys(collection?.properties ?? {}).find(propertyName =>
+		Boolean(collection?.properties?.[propertyName]['<key>']),
+	);
+
+	const insertionPath = getKeySpaceReference(collection);
+	const isKeyGeneratedWithFakerFunction = collection?.properties?.[keyPropertyName]?.fakerFunction;
+	const parseJsonData = JSON.parse(jsonData);
+	const keyFakedValue = isKeyGeneratedWithFakerFunction ? parseJsonData[keyPropertyName] : '';
+	const jsonDataBody = _.omit(parseJsonData, [keyPropertyName]);
```

Comment:
Or you can even replace _.omit by destructuring with rest.
`const { keyPropertyName: keyProperty, ...jsonDataBody } = parseJsonData`
But this change isn't required.